pingcap/tidb
# *: preload load frequent use variable

### What problem does this PR solve? 

fixes #10388, preload [JDBC](https://github.com/mysql/mysql-connector-j/blob/34cbc6bc61f72836e26327537a432d6db7c77de6/src/main/core-impl/java/com/mysql/cj/NativeSession.java) used variable into memory as default value 

this will improve to java based application that frequent establish new connections to TiDB.

### What is changed and how it works?

preload those variable:

```
	variable.InitConnect,
	variable.TxnIsolation,
	variable.TxReadOnly,
	variable.TransactionIsolation,
	variable.TransactionReadOnly,
	variable.NetBufferLength,
	variable.QueryCacheType,
	variable.QueryCacheSize,
	variable.CharacterSetServer,
	variable.AutoIncrementIncrement,
	variable.CollationServer,
	variable.NetWriteTimeout,
```

so, they can preload into session.

change `MaxAllowedPacket` as global + session scope  https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet

also change `InitConnect`, `QueryCacheSize` from global scope to global + session scope, this is not same with mysql, but doesn't break behave. why we need change this is TiDB will access kv every time if it's a global only variable, but for this two variable we no need do that in performance reason.


### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 
```
add log in GetGlobalSys method and start a jdbc app and connect to tidb and should not see any log output any more.
```

Code changes

 - impl change

Side effects

 - N/A

Related changes

 - N/A


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/10463)
